### PR TITLE
Cache the XRandR extension version

### LINKED ISSUE #3335: Abnormally slow startup for X11
cc: https://github.com/alacritty/alacritty/issues/7472

It takes almost 3 seconds to create a window, which is frankly absurd.

### PR DESCRIPTION
**Pull request by notgull**
Maybe this will fix #3335

- [x] Tested on all platforms changed
- [ ] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users
- [x] Updated documentation to reflect any user-facing changes, including notes of platform-specific behavior
- [x] Created or updated an example program if it would help users understand this functionality
- [x] Updated [feature matrix](https://github.com/rust-windowing/winit/blob/master/FEATURES.md), if new features were added or implemented

### CODE REVIEW

**Comment by notgull** [comment]
This doesn't look like it fixes the issue, but it's probably good to do anyways

**Comment by kchibisov** [comment]
Do we need this?

**Comment by notgull** [comment]
It's probably nice to have. But I need to rebase it first

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,7 +11,9 @@ Unreleased` header.
 
 # Unreleased
 
+
```
Comment (kchibisov) [inline CHANGELOG.md:14]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,7 +11,9 @@ Unreleased` header.
...
 - On X11, reduce the amount of time spent fetching screen resources.
+- Cache the XRandR version before calling `available_monitors()`.
```
Comment (kchibisov) [inline CHANGELOG.md:16]: Not relevant, since the other entry kind of accounts for it already.

```suggestion
```

**Review by kchibisov — CHANGES_REQUESTED** [review]

**Review by kchibisov — APPROVED** [review]

**Pull request merged**